### PR TITLE
Feat/chapter export

### PR DESCRIPTION
**Pull request by KillerX**
This is missing the integration into Temporal but it is working otherwise. I'll extract the HTTP parts and set up some mock testing, but feel free to review and comment. Especially if you have good ideas for names/what parts could be wrapped into functions.

Related: https://github.com/bcc-code/bcc-media-platform/issues/511

### CODE REVIEW

File: services/vidispine/chapter.go [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+	allChapters := map[string]*MetadataResult{}
+
+	for _, clip := range exportData.Clips {
+		if _, ok := metaCache[clip.VXID]; !ok {
+			meta, err := c.GetMetadata(clip.VXID)
+			if err != nil {
+				return nil, err
+			}
+			metaCache[clip.VXID] = meta
+		}
```
Comment (fredrikved) [inline services/vidispine/chapter.go:92]: I think the vidispine client is initialized per activity/request, so this cache could probably be stored on the client struct and moved out?

File: services/vidispine/chapter.go [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+
+	chapters := []Chapter{}
+	for _, data := range allChapters {
+		chapters = append(chapters, MetaToChapter(data))
+	}
+
+	return chapters, nil
+}
+
+func MetaToChapter(meta *MetadataResult) Chapter {
```
Comment (fredrikved) [inline services/vidispine/chapter.go:144]: Maybe these functions and types should be private?

File: services/vidispine/export.go [hunk 3]
```diff
@@ -408,3 +407,33 @@ func (c *Client) GetDataForExport(itemVXID string) (*ExportData, error) {
 
 	return &out, nil
 }
+
+// ConvertFromClipTCTimeSpaceToSequenceRelativeTimeSpace ain't this a nice name?
+func ConvertFromClipTCTimeSpaceToSequenceRelativeTimeSpace(clip *Clip, chapter *MetadataResult, tcStart float64) *MetadataResult {
```
Comment (fredrikved) [inline services/vidispine/export.go:412]: Amazing function name, but making it private makes it less noticeable :). Not that I wouldn't want to see it of course.

File: services/vidispine/chapter.go (hunk 1 above)
Comment (KillerX) [reply services/vidispine/chapter.go]: Yes I wanted it to be at most cached for one request.

**Review by fredrikved — APPROVED** [review]

**Pull request merged**